pkg/platform/github.go

```diff
@@ -150,6 +154,97 @@ func (g *GitHub) AssignReviewers(ctx context.Context, input *AssignReviewersInpu
 	return &result, nil
 }
 
+type pullRequestReview struct {
+	Author struct {
+		Login string
+	}
+	State string
+}
+
+type latestApproverQuery struct {
+	Repository struct {
+		PullRequest struct {
+			LatestReviews struct {
+				Nodes []pullRequestReview
+			} `graphql:"latestReviews(first: 100)"`
+		} `graphql:"pullRequest(number: $number)"`
+	} `graphql:"repository(owner: $owner, name: $repo)"`
+}
+
+type member struct {
+	Login string
+}
+type team struct {
+	Name    string
+	Members struct {
+		Nodes []member
+	} `graphql:"members(first: 100)"`
+}
+type organizationTeamsAndMembershipsQuery struct {
+	Organization struct {
+		Teams struct {
+			Nodes []team
+		} `graphql:"teams(first: 100)"`
+	} `graphql:"organization(login: $owner)"`
+}
+
+// GetLatestApprovers retrieves the users whose latest review for a pull request
+// is an approval. It also returns the teams and subteams that the user
+// approvers are members of to indicate approval on behalf of those teams. Note,
+// a comment following a previous approval by the same user will still keep the
+// APPROVED state. However, if a reviewer previously approved the PR and
+// requests changes/dismisses the review, then the approval is not counted.
+func (g *GitHub) GetLatestApprovers(ctx context.Context) (*GetLatestApproversResult, error) {
+	logger := logging.FromContext(ctx)
+	logger.DebugContext(ctx, "querying latest approvers")
+
+	var approversQuery latestApproverQuery
+	if err := g.graphqlClient.Query(ctx, &approversQuery, map[string]any{
+		"owner":  githubv4.String(g.cfg.GitHubOwner),
+		"repo":   githubv4.String(g.cfg.GitHubRepo),
+		"number": githubv4.Int(g.cfg.GitHubPullRequestNumber),
+	}); err != nil {
+		return nil, fmt.Errorf("failed to query latest approvers: %w", err)
+	}
+
+	// Explicitly sets the default Users and Teams to empty slices. If these are
+	// not explicitly provided to OPA, then the policy result may be incorrect.
+	result := &GetLatestApproversResult{
+		Teams: []string{},
+		Users: []string{},
+	}
+	hasApproved := make(map[string]struct{}, len(approversQuery.Repository.PullRequest.LatestReviews.Nodes))
+	for _, review := range approversQuery.Repository.PullRequest.LatestReviews.Nodes {
+		if review.State == "APPROVED" {
+			result.Users = append(result.Users, review.Author.Login)
+			hasApproved[review.Author.Login] = struct{}{}
+		}
+	}
+
+	var teamQuery organizationTeamsAndMembershipsQuery
+	if err := g.graphqlClient.Query(ctx, &teamQuery, map[string]any{
```

Comment:
followup: should we paginate this? I would guess some orgs might have over 100 teams, users, etc

Example: https://github.com/abcxyz/github-metrics-aggregator/blob/main/pkg/review/commit_review_status.go